allennlp/data/vocabulary.py

```diff
@@ -223,6 +223,7 @@ def __init__(
         self._padding_token = padding_token if padding_token is not None else DEFAULT_PADDING_TOKEN
         self._oov_token = oov_token if oov_token is not None else DEFAULT_OOV_TOKEN
 
+        self._last_save_directory = None
```

Comment:
This needs a type annotation (`self._last_save_directory: str = None`) to make mypy happy.